frontend/src/components/pages/Register.js

```diff
@@ -0,0 +1,112 @@
+import axios from 'axios';
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+const Register = () => {
+  const history = useHistory();
+
+  const [registerInput, setRegister] = useState({
+    name: '',
+    username: '',
+    email: '',
+    password: '',
+    error_list: [],
+  });
+
+  const handleInput = (e) => {
+    e.persist();
+    setRegister({ ...registerInput, [e.target.name]: e.target.value });
+  };
+
+  const registerSubmit = (e) => {
+    e.preventDefault();
+
+    const data = {
+      name: registerInput.name,
+      username: registerInput.username,
+      email: registerInput.email,
+      password: registerInput.password,
+    };
+    axios.get('/sanctum/csrf-cookie').then((response) => {
+      axios.post(`/api/register`, data).then((res) => {
+        if (res.data.status === 200) {
+          localStorage.setItem('auth_token', res.data.token);
+          history.push('/');
+          localStorage.setItem('auth_name', res.data.username);
+        } else {
+          setRegister({
+            ...registerInput,
+            error_list: res.data.validation_errors,
+          });
+        }
+      });
```

Comment:
I want you to have separate file for this one. "Separating api calling into a different files makes our code organize and easy to troubleshoot when we trace some problems regarding about apis.